metadata/src/test/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -371,8 +371,21 @@ public void register(MetaLogListener listener) throws Exception {
 
     @Override
     public long scheduleWrite(long epoch, List<ApiMessageAndVersion> batch) {
-        return shared.tryAppend(nodeId, leader.epoch(), new LocalRecordBatch(
-            batch.stream().map(r -> r.message()).collect(Collectors.toList())));
+        return scheduleAtomicWrite(epoch, batch);
+    }
+
+    @Override
+    public long scheduleAtomicWrite(long epoch, List<ApiMessageAndVersion> batch) {
```

Comment:
Same comment for the other LocalLogManager. Hmm 🤔 why do we have two of these?

Reply:
Yeah. I noticed this too. Not clear if this was on purpose. I say we fix this in a future PR.